packages/integration-objects/srv/validations/ValidateNonPrimaryFieldsForRates.js

```diff
@@ -0,0 +1,52 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+const util = require('@sap-cloud-sdk/util');
+const { ValidationError } = require('../exceptions/validation-error');
+const { logger } = require('../logging/Logger');
+const Constants = require('../utils/Constants');
+const ErrorStatuses = require('../utils/ErrorStatuses');
+const RateExtensionConstants = require('../utils/RateExtensionConstants');
+
+function validateNonPrimaryKeyFieldsForCurrencyExchangeRate(data) {
+  validateExchangeRateValue(data);
+  setRateValueIndirect(data);
+  validateFromCurrencyFactor(data);
+  validateToCurrencyFactor(data);
+}
+
+module.exports = {
+  validateNonPrimaryKeyFieldsForCurrencyExchangeRate
+};
+
+function validateToCurrencyFactor(data) {
```

Comment:
the methods `validateToCurrencyFactor` and `validateFromCurrencyFactor` are the same. Consider making then one method: `validateNumberInBounds(value,minMaxValue,errorMessage)`. It also seems strange that both from and to currency factors have to be below 1? You would assume that if one is above the other one hase to below one, right?

Add then a unit test for the validateNumberInBounds method.

Reply:
I have updated this as an issue to be taken up in the next release.
Both the currency factors have to be checked for below 1 as per the Java code. I reverified it too.

Reply:
ok